Storage/container/one_to_many.h

```diff
@@ -0,0 +1,261 @@
+    DoAdd(std::make_pair(row, col), e.data);
+  }
+  void operator()(const T_DELETE_EVENT& e) { DoErase(std::make_pair(e.key.first, e.key.second)); }
+
+  template <typename T_MAP>
+  struct Iterator final {
+    using T_ITERATOR = typename T_MAP::const_iterator;
+    using T_KEY = typename T_MAP::key_type;
+    T_ITERATOR iterator_;
+    explicit Iterator(T_ITERATOR iterator) : iterator_(iterator) {}
+    void operator++() { ++iterator_; }
+    bool operator==(const Iterator& rhs) const { return iterator_ == rhs.iterator_; }
+    bool operator!=(const Iterator& rhs) const { return !operator==(rhs); }
+    sfinae::CF<T_KEY> key() const { return iterator_->first; }
+    const T& operator*() const { return *iterator_->second; }
+    const T* operator->() const { return iterator_->second; }
+  };
+
+  template <typename INNER_MAP>
+  struct InnerAccessor final {
+    using INNER_KEY = typename INNER_MAP::key_type;
+    const INNER_MAP& map_;
+
+    InnerAccessor(const INNER_MAP& map) : map_(map) {}
+
+    bool Empty() const { return map_.empty(); }
+    size_t Size() const { return map_.size(); }
+
+    bool Has(const INNER_KEY& x) const { return map_.find(x) != map_.end(); }
+
+    Iterator<INNER_MAP> begin() const { return Iterator<INNER_MAP>(map_.cbegin()); }
+    Iterator<INNER_MAP> end() const { return Iterator<INNER_MAP>(map_.cend()); }
+  };
+
+  template <typename OUTER_MAP>
+  struct OuterAccessor final {
+    using OUTER_KEY = typename OUTER_MAP::key_type;
+    using INNER_MAP = typename OUTER_MAP::mapped_type;
+    const OUTER_MAP& map_;
+
+    struct OuterIterator final {
+      using T_ITERATOR = typename OUTER_MAP::const_iterator;
+      T_ITERATOR iterator;
+      explicit OuterIterator(T_ITERATOR iterator) : iterator(iterator) {}
+      void operator++() { ++iterator; }
+      bool operator==(const OuterIterator& rhs) const { return iterator == rhs.iterator; }
+      bool operator!=(const OuterIterator& rhs) const { return !operator==(rhs); }
+      sfinae::CF<OUTER_KEY> key() const { return iterator->first; }
+      InnerAccessor<INNER_MAP> operator*() const { return InnerAccessor<INNER_MAP>(iterator->second); }
+    };
+
+    explicit OuterAccessor(const OUTER_MAP& map) : map_(map) {}
+
+    bool Empty() const { return map_.empty(); }
+    size_t Size() const { return map_.size(); }
+
+    bool Has(const OUTER_KEY& x) const { return map_.find(x) != map_.end(); }
+
+    ImmutableOptional<InnerAccessor<INNER_MAP>> operator[](OUTER_KEY key) const {
+      const auto iterator = map_.find(key);
+      if (iterator != map_.end()) {
+        return std::move(std::make_unique<InnerAccessor<INNER_MAP>>(iterator->second));
+      } else {
+        return nullptr;
+      }
+    }
+
+    OuterIterator begin() const { return OuterIterator(map_.cbegin()); }
+    OuterIterator end() const { return OuterIterator(map_.cend()); }
+  };
+
+  const OuterAccessor<T_FORWARD_MAP> Rows() const { return OuterAccessor<T_FORWARD_MAP>(forward_); }
```

Comment:
Why not rename `OuterAccessor` and `InnerAccessor` into `RowsAccessor` and `ElementsAccessor`?